solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1365,6 +1382,32 @@ public void refreshAndWatch() {
         log.error("Unwatched collection: [{}]", coll, e);
       }
     }
+
+    private void refreshAndWatchChildren() throws KeeperException, InterruptedException {
```

Comment:
Some of the code in this method duplicates code from `refreshAndWatch` from where it is called. Is there a way to refactor so that code common to both only appears once?